### PR TITLE
Fix samples

### PR DESCRIPTION
**Pull request by APrigarina**
This PR fixes errors in some samples

### CODE REVIEW

File: samples/python/hist.py [hunk 1]
```diff
@@ -46,9 +46,9 @@ def hist_lines(im):
         im = cv.cvtColor(im,cv.COLOR_BGR2GRAY)
     hist_item = cv.calcHist([im],[0],None,[256],[0,256])
     cv.normalize(hist_item,hist_item,0,255,cv.NORM_MINMAX)
-    hist=np.int32(np.around(hist_item))
+    hist = np.around(hist_item)
     for x,y in enumerate(hist):
-        cv.line(h,(x,0),(x,y),(255,255,255))
+        cv.line(h,(int(x),0),(int(x),int(y)),(255,255,255))
```
Comment (alalek) [inline samples/python/hist.py:51]: What was type of `x` here?

File: samples/python/video_v4l2.py [hunk 2]
```diff
@@ -30,11 +30,11 @@ def decode_fourcc(v):
     color = (0, 255, 0)
 
     cap = cv.VideoCapture(0)
-    cap.set(cv.CAP_PROP_AUTOFOCUS, False)  # Known bug: https://github.com/opencv/opencv/pull/5474
+    cap.set(cv.CAP_PROP_AUTOFOCUS, 0)  # Known bug: https://github.com/opencv/opencv/pull/5474
 
     cv.namedWindow("Video")
 
-    convert_rgb = True
+    convert_rgb = 1
```
Comment (alalek) [inline samples/python/video_v4l2.py:37]: It makes sense to replace passing of `convert_rgb` argument instead of changing its type:

`convert_rgb` => `1 if convert_rgb else 0`

**Review by alalek — COMMENTED** [review]
Thank you!

File: samples/python/camera_calibration_show_extrinsics.py [hunk 3]
```diff
@@ -188,7 +188,7 @@ def main():
 
     fig = plt.figure()
     ax = fig.gca(projection='3d')
-    ax.set_aspect("equal")
+    ax.set_aspect("auto")
```
Comment (catree) [inline samples/python/camera_calibration_show_extrinsics.py:191]: @APrigarina 

What is the reason of this change?

Before:

image

After:

image

Comment (APrigarina) [reply samples/python/camera_calibration_show_extrinsics.py:191]: @catree, this is because of matplotlib bug in versions >=3.1.0 on Windows. The picture depends on window size, the graph is still valid. On latest versions of matlotlib the picture doesn't change with window resizing
image

Comment (catree) [reply samples/python/camera_calibration_show_extrinsics.py:191]: @APrigarina 

Ok, I will trust your change. Any possibility to make it looks good on any platform regardless of the Matplotlib version would be welcome.

File: samples/python/hist.py (hunk 1 above)
Comment (alalek) [reply samples/python/hist.py]: what was wrong with type of `y` too?

Comment (APrigarina) [reply samples/python/hist.py]: it is numpy array of one float element

**Comment by alalek** [comment]
@APrigarina Please rebase patch on latest master branch (there was merged 3.4 variant of this patch)

File: samples/python/hist.py [hunk 4]
```diff
@@ -46,9 +46,9 @@ def hist_lines(im):
         im = cv.cvtColor(im,cv.COLOR_BGR2GRAY)
     hist_item = cv.calcHist([im],[0],None,[256],[0,256])
     cv.normalize(hist_item,hist_item,0,255,cv.NORM_MINMAX)
-    hist=np.int32(np.around(hist_item))
+    hist = np.around(hist_item)
     for x,y in enumerate(hist):
-        cv.line(h,(x,0),(x,y[0]),(255,255,255))
+        cv.line(h,(x,0),(x,int(y)),(255,255,255))
```
Comment (alalek) [inline samples/python/hist.py:51]: Why we should have different code on 3.4 and master branch?

Comment (APrigarina) [reply samples/python/hist.py:51]: on 3.4 branch there was array of one integer element, on master branch - of one float

Comment (alalek) [reply samples/python/hist.py:51]: There was `hist=np.int32(...)` above.

Comment (APrigarina) [reply samples/python/hist.py:51]: yes, I missed it, thanks

**Review by alalek — APPROVED** [review]
Thank you :+1:
